### ISSUE TITLE
Bug in our visibility tracking for non-instance methods

### ISSUE THREAD

**Issue by troessner** [troessner/reek#792]
#724 uncovered a bug in our visibility tracking in regards to non-instance methods.

The following spec illustrates this bug:

```Ruby
    it 'reports the visibility of class methods properly' do
      src = <<-EOS
        class Car
          private
          def self.drive; end
        end
      EOS

      syntax_tree = Reek::Source::SourceCode.from(src).syntax_tree

      context_tree = Reek::ContextBuilder.new(syntax_tree).send(:context_tree)

      singleton_method = context_tree.children.first.children.first
      expect(singleton_method.visibility).to eq(:public)
    end
```

The output of this:

```
  1) Reek::Context::CodeContext enumerating syntax elements reports the visibility of class methods properly
     Failure/Error: expect(singleton_method.visibility).to eq(:public)

       expected: :public
            got: :private
```

@mvz || @chastell any of you got time to look into this?

**Comment by mvz** [comment]
I may be able to look at it tomorrow, as a fun Christmas activity :smile:.

**Comment by mvz** [comment]
Fixed in #796.
